core/src/test/java/com/linecorp/armeria/server/auth/AuthServiceTest.java

```diff
@@ -429,11 +434,23 @@ void shouldPeelRedundantAuthorizerExceptions() throws Exception {
                            .onFailure((delegate, ctx, req, cause) -> {
                                causeRef.set(cause);
                                return HttpResponse.of(HttpStatus.FORBIDDEN);
-                           }).build((ctx, req) -> HttpResponse.of("OK"));
+                           })
+                           .build((ctx, req) -> HttpResponse.of("OK"));
+
+        // Create a meter registry and set a timer to the service.
+        final MeterRegistry meterRegistry = new SimpleMeterRegistry();
+        final Field timerField = AuthService.class.getDeclaredField("timer");
+        timerField.setAccessible(true);
```

Comment:
I don't think we need reflection + what do you think of creating a dedicated test for recording metrics instead

```java
diff --git a/core/src/test/java/com/linecorp/armeria/server/auth/AuthServiceTest.java b/core/src/test/java/com/linecorp/armeria/server/auth/AuthServiceTest.java
index cb65d5c8b..580f6f7de 100644
--- a/core/src/test/java/com/linecorp/armeria/server/auth/AuthServiceTest.java
+++ b/core/src/test/java/com/linecorp/armeria/server/auth/AuthServiceTest.java
@@ -36,6 +36,7 @@ import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
 import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
 import org.apache.hc.client5.http.impl.classic.HttpClients;
 import org.assertj.core.util.Strings;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
@@ -43,6 +44,7 @@ import com.google.common.collect.ImmutableMap;
 
 import com.linecorp.armeria.client.BlockingWebClient;
 import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
 import com.linecorp.armeria.common.HttpHeaderNames;
 import com.linecorp.armeria.common.HttpHeaders;
 import com.linecorp.armeria.common.HttpMethod;
@@ -83,10 +85,14 @@ class AuthServiceTest {
 
     private static final AsciiString CUSTOM_TOKEN_HEADER = HttpHeaderNames.of("X-Custom-Authorization");
 
+    private static final SimpleMeterRegistry meterRegistry = new SimpleMeterRegistry();
+
     @RegisterExtension
     static final ServerExtension server = new ServerExtension() {
         @Override
         protected void configure(ServerBuilder sb) throws Exception {
+            sb.meterRegistry(meterRegistry);
+
             final HttpService ok = new AbstractHttpService() {
                 @Override
                 protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
@@ -227,6 +233,11 @@ class AuthServiceTest {
         }
     }
 
+    @BeforeEach
+    void beforeEach() {
+        meterRegistry.clear();
+    }
+
     @Test
     void testBasicAuth() throws Exception {
         final BlockingWebClient webClient = WebClient.builder(server.httpUri())
@@ -422,6 +433,15 @@ class AuthServiceTest {
         }
     }
 
+    @Test
+    void shouldRecordMetrics() throws Exception {
+        assertThat(MoreMeters.measureAll(meterRegistry)).containsEntry("armeria.server.auth#count", 0.0);
+        AggregatedHttpResponse res = server.blockingWebClient(cb -> cb.auth(AuthToken.ofBasic("brown", "cony")))
+                                           .get("/on_failure_exception");
+        assertThat(res.status().code()).isEqualTo(200);
+        assertThat(MoreMeters.measureAll(meterRegistry)).containsEntry("armeria.server.auth#count", 1.0);
+    }
+
     @Test
     void shouldPeelRedundantAuthorizerExceptions() throws Exception {
         final AtomicReference<Throwable> causeRef = new AtomicReference<>();

```